docs/design/014-did-doc/README.md

```diff
@@ -0,0 +1,590 @@
+# DID Doc Support in Indy
+
+## Purpose
+
+The purpose of this document is to provide vision how we can integrate w3c DIDDoc specification to Indy
+infrastructure and potentially achieve some level of interoperability with 3d party solutions that also
+follow this specification.
+
+## Use Cases
+
+As an identity owner i want to:
+
+1. create a new DID and 1 or multiple related keys
+1. publish corresonded DID Doc to Public Ledger if i have sufficient rights
+1. provide DID Doc to 3d party (Trust Ancher) for publishing to Public Ledger if i have no sufficient rights
+1. publish new DID Doc to microledger
+1. add second key to the DID DOC from a different edge device
+1. rotate a key from an edge device
+1. change authentication/authorization policies from one of the devices
+1. change service endpoints from an edge device
+1. support of multiple endpoints, each working with multiple edge devices
+1. know the recent key for this edge device to be used for signature and A2A communication using this edge device
```

Comment:
Not only "for this edge device" but for any edge device of this relationship.

Reply:
@lovesh @dhh1128  @ashcherbakov  I am not sure that it is necessary and feasible to get complete Edge devices topology and key/device mapping information from DID Doc. Some applications can provide this information, but it is application level, not libindy level.  What edge device know:

- The list of keys related to this DID stored in wallet
- Device can resolve the latest version of DID Doc and based on this information understand what local keys can be used to perform DID Doc modifications and what keys can be used for A2A communication

But It can be impossible to know on what devices other keys stored without getting some additional information from application user.

Reply:
I agree with Slava that Alice's edge device doesn't need to know all keys for all other Alice's edge device. A good example is a static agent (an IoT device) that need to know only the key it owns.